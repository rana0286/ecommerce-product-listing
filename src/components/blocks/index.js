import ProductsSizeFilter from './ProductSizeFilter/ProductSizeFilter';
import ProductsListHeading from './ProductsListHeading/ProductListHeading';
import ProductDetailsCard from './ProductDetailsCard/ProductDetailsCard';

export {
  ProductsSizeFilter,
  ProductsListHeading,
  ProductDetailsCard,
};
