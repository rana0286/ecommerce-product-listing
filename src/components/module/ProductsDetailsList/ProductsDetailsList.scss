.product-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  list-style: none;
  margin-bottom: 30px;
  padding-left: 0;
}
