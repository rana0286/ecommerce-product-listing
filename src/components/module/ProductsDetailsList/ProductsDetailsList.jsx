import React from 'react';
import PropTypes from 'prop-types';
import { ProductDetailsCard } from '../../blocks';

import './ProductsDetailsList.scss';
/**
  * @desc ProductsList component can be used to render render list of products.
  *  @productsList : contains array of products information
*/


const ProductsDetailsList = ({ productsList }) => (
  <ul className="product-list">
    {productsList.map(product => <ProductDetailsCard product={product} key={product.index} />)}
  </ul>
);

ProductsDetailsList.propTypes = {
  productsList: PropTypes.arrayOf(
    PropTypes.shape({
      productImage: PropTypes.string.isRequired,
      productName: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
      isExclusive: PropTypes.bool.isRequired,
      isSale: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};

export default ProductsDetailsList;
