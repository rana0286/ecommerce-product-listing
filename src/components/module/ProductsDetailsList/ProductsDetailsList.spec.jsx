import React from 'react';
import { mount} from 'enzyme';
import ProductsDetailsList from './ProductsDetailsList';
import { ProductDetailsCard } from '../../blocks';
import renderer from 'react-test-renderer';

describe('src/components/module/ProductsDetailsList', () => {
  const produtList = [
    {
      "size": [
        "XS",
        "S",
        "L",
        "XL"
      ],
      "productName": "Striped shirt",
      "productImage": "product-1.jpg",
      "price": "$18.88",
      "isExclusive": false,
      "isSale": true,
      "index": 0
    },
    {
      "size": [
        "XS",
        "S"
      ],
      "productName": "Denim shirt",
      "productImage": "product-2.jpg",
      "price": "$25.44",
      "isExclusive": false,
      "isSale": false,
      "index": 1
    },
    {
      "size": [
        "S",
        "M"
      ],
      "productName": "Plain cotton t-shirt",
      "productImage": "product-3.jpg",
      "price": "$12.93",
      "isExclusive": true,
      "isSale": false,
      "index": 2
    }
  ];

  const componentRenderer = (props) => <ProductsDetailsList {...props}/>;
  it('ProductsDetailsList component should render', () => {
    const tree = renderer.create(componentRenderer({productsList: produtList})).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('ProductsDetailsList should render 3 product card', () => {
    const component = mount(componentRenderer({productsList: produtList}));
    expect(component.find(ProductDetailsCard).length).toBe(3);
  });
});
