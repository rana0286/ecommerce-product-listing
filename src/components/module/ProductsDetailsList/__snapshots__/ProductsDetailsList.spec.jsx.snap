// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/module/ProductsDetailsList ProductsDetailsList component should render 1`] = `
<ul
  className="product-list"
>
  <li
    className="card"
  >
    <img
      alt="Striped shirt"
      className="image"
      src="./assets/img/products/product-1.jpg"
    />
    <div
      className="product"
    >
      <div
        className="tag"
      >
        <div
          className="tag--sale"
        >
          Sale
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="description__name"
        >
          Striped shirt
        </div>
        <div
          className="description__price"
        >
          $18.88
        </div>
      </div>
    </div>
  </li>
  <li
    className="card"
  >
    <img
      alt="Denim shirt"
      className="image"
      src="./assets/img/products/product-2.jpg"
    />
    <div
      className="product"
    >
      <div
        className="tag"
      />
      <div
        className="description"
      >
        <div
          className="description__name"
        >
          Denim shirt
        </div>
        <div
          className="description__price"
        >
          $25.44
        </div>
      </div>
    </div>
  </li>
  <li
    className="card"
  >
    <img
      alt="Plain cotton t-shirt"
      className="image"
      src="./assets/img/products/product-3.jpg"
    />
    <div
      className="product"
    >
      <div
        className="tag"
      >
        <div
          className="tag--exclusive"
        >
          Exclusive
        </div>
      </div>
      <div
        className="description"
      >
        <div
          className="description__name"
        >
          Plain cotton t-shirt
        </div>
        <div
          className="description__price"
        >
          $12.93
        </div>
      </div>
    </div>
  </li>
</ul>
`;
